spec/controllers/infra_networking_controller_spec.rb

```diff
@@ -1,27 +1,120 @@
+  let(:host) { FactoryGirl.create(:host, :name => 'test_host1') }
+  let(:ems_vmware) { FactoryGirl.create(:ems_vmware, :name => 'test_vmware') }
+  let(:cluster) { FactoryGirl.create(:cluster, :name => 'test_cluster') }
+  let(:user) { FactoryGirl.create(:user) }
+  let(:classification) { FactoryGirl.create(:classification, :name => "department", :description => "Department") }
+  let(:tag1) { FactoryGirl.create(:classification_tag, :name   => "tag1", :parent => classification) }
+  let(:tag2) { FactoryGirl.create(:classification_tag, :name   => "tag2", :parent => classification) }
 
-  before { stub_user(:features => :all) }
+  before do
+    stub_user(:features => :all)
+    FactoryGirl.create(:tagging, :tag => tag1.tag, :taggable => switch)
+    FactoryGirl.create(:tagging, :tag => tag2.tag, :taggable => switch)
+  end
 
-  describe '#tree_select' do
+  describe 'render_views' do
     render_views
 
-    context 'switch' do
-      let(:ems) { FactoryGirl.create(:ems_vmware) }
-      let(:cluster) { FactoryGirl.create(:ems_cluster, :ems_id => ems.id) }
-      let(:host) { FactoryGirl.create(:host_vmware, :ems_id => ems.id, :ems_cluster => cluster) }
-      let(:switch) do
-        sw = FactoryGirl.create(:switch_vmware, :ems_id => ems.id)
-        FactoryGirl.create(:host_switch, :host => host, :switch => sw)
-        sw
+    before do
+      EvmSpecHelper.create_guid_miq_server_zone
+    end
+
+    describe '#explorer' do
+      before do
+        session[:settings] = {:views => {}, :perpage => {:list => 5}}
       end
 
-      it 'renders the network switch center toolbar' do
-        nodeid = [ems, cluster, switch].map { |item| TreeNode.new(item).key }.join('_')
-        expect(ApplicationHelper::Toolbar::InfraNetworkingCenter).to receive(:definition).and_call_original.at_least(:once)
-        post :tree_select, :params => { :id => nodeid }
+      it 'can render the explorer' do
+        session[:sb] = {:active_accord => :infra_networking_accord}
+        seed_session_trees('switch', :infra_networking_tree, 'root')
+        get :explorer
+        expect(response.status).to eq(200)
+        expect(response.body).to_not be_empty
+      end
+
+      it 'shows a switch in the list' do
+        session[:sb] = {:active_accord => :infra_networking_accord}
+        seed_session_trees('switch', :infra_networking_tree, 'root')
+
+        get :explorer
+        expect(response.body).to include("modelName: 'Switch'")
+        expect(response.body).to include("activeTree: 'infra_networking_tree'")
+        expect(response.body).to include("gtlType: 'list'")
+        expect(response.body).to include("isExplorer: 'true' === 'true' ? true : false")
+        expect(response.body).to include("showUrl: '/infra_networking/x_show/'")
+      end
+    end
+
+    describe "#tree_select" do
+      [
+        ['All Distributed Switches', :infra_networking_tree],
+      ].each do |elements, tree|
```

Comment:
After looking at this for 15 minutes :rofl: I think this `each` here is not necessary, it's a single element array. Just move the values into the spec and omit the variables.